controllers/nodehealthcheck_controller.go

```diff
@@ -374,7 +374,7 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 		// check if we need to alert about a very old remediation CR
 		remediationCRs, err := resourceManager.ListRemediationCRs(utils.GetAllRemediationTemplates(nhc), func(cr unstructured.Unstructured) bool {
-			return cr.GetName() == node.GetName() && resources.IsOwner(&cr, nhc)
+			return utils.GetNodeNameFromCR(cr) == node.GetName() && resources.IsOwner(&cr, nhc)
```

Comment:
this fixes `[FAIL] Node Health Check CR Reconciliation with a single escalating remediation with multiple same kind support when an old remediation cr exists [It] an alert flag is set on remediation cr`